### PR TITLE
chore: need this change on default branch

### PR DESCRIPTION
**Pull request by peternhale**
@W-13190794@

### What does this PR do?
sets up a e2e test that uses yourkit profiling

POC -

### CODE REVIEW

File: .github/workflows/e2ePerf.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            salesforcedx-vscode-automation-tests/package-lock.json
+      - name: Clone automation tests
+        uses: actions/checkout@v3
+        with:
+          repository: forcedotcom/salesforcedx-vscode-automation-tests
+          path: salesforcedx-vscode-automation-tests
+          ref: ${{ inputs.automationBranch }}
+      - name: Install Extension Dependencies
+        run: |
+          npm install
```
Comment (CristiCanizales) [inline .github/workflows/e2ePerf.yml:53]: I think we also need to run npm run compile here ?

Comment (randi274) [reply .github/workflows/e2ePerf.yml:53]: Yes - that was a missing step for us previously!

**Review by randi274 — CHANGES_REQUESTED** [review]
@peternhale looks pretty straightforward! We do like to test the workflows first - can you create a fork for VS Code to run it, and then add a sample action run to the description? The PR can be opened from the fork or we can keep this one open, as long as the files are identical.

Example PR: https://github.com/forcedotcom/salesforcedx-vscode/pull/4811

File: .github/workflows/e2ePerf.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+name: End to End Tests (Performance)
+on:
+  workflow_dispatch:
+    inputs:
+      automationBranch:
+        description: 'Set the branch to use for automation tests'
+        required: false
+        default: 'phale/perf-poc'
```
Comment (randi274) [inline .github/workflows/e2ePerf.yml:8]: And then this should probably be 'develop'

**Pull request closed without merging**